src/librustc/middle/stability.rs

```diff
@@ -597,8 +597,31 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                                            feature.as_str(), &r),
                     None => format!("use of unstable library feature '{}'", &feature)
                 };
-                emit_feature_err(&self.sess.parse_sess, &feature.as_str(), span,
-                                 GateIssue::Library(Some(issue)), &msg);
+
+
+                let msp: MultiSpan = span.into();
+                let cm = &self.sess.parse_sess.codemap();
+                let span_key =
+                    msp.primary_span().and_then(|sp:Span|
+                        if sp != DUMMY_SP {
```

Comment:
Clean up the formatting for this entire block to avoid as having as much left padding.